source/Calamari/Kubernetes/ResourceStatus/ResourceStatusReportExecutor.cs

```diff
@@ -1,160 +1,48 @@
+#if !NET40
 using System;
 using System.Collections.Generic;
-using System.IO;
 using System.Linq;
-using Calamari.Common.Plumbing.FileSystem;
-using Calamari.Common.Plumbing.Logging;
+using System.Threading;
 using Calamari.Common.Plumbing.Variables;
-using Calamari.Kubernetes.Integration;
 using Calamari.Kubernetes.ResourceStatus.Resources;
 
 namespace Calamari.Kubernetes.ResourceStatus
 {
     public class ResourceStatusReportExecutor
     {
-        private const int PollingIntervalSeconds = 2;
-
         private readonly IVariables variables;
-        private readonly ILog log;
-        private readonly ICalamariFileSystem fileSystem;
-        private readonly IResourceStatusChecker statusChecker;
-        private readonly Kubectl kubectl;
+        private readonly RunningResourceStatusCheck.Factory runningResourceStatusCheckFactory;
 
         public ResourceStatusReportExecutor(
             IVariables variables,
-            ILog log,
-            ICalamariFileSystem fileSystem,
-            IResourceStatusChecker statusChecker,
-            Kubectl kubectl)
+            RunningResourceStatusCheck.Factory runningResourceStatusCheckFactory)
         {
             this.variables = variables;
-            this.log = log;
-            this.fileSystem = fileSystem;
-            this.statusChecker = statusChecker;
-            this.kubectl = kubectl;
+            this.runningResourceStatusCheckFactory = runningResourceStatusCheckFactory;
         }
 
-        public void ReportStatus(string workingDirectory)
+        public IRunningResourceStatusCheck Start(IEnumerable<ResourceIdentifier> resources)
         {
-            var defaultNamespace = variables.Get(SpecialVariables.Namespace, "default");
-            // When the namespace on a target was set and then cleared, it's going to be "" instead of null
-            if (string.IsNullOrEmpty(defaultNamespace))
-            {
-                defaultNamespace = "default";
-            }
-            var timeoutSeconds = variables.GetInt32(SpecialVariables.Timeout) ?? 0;
-            var waitForJobs = variables.GetFlag(SpecialVariables.WaitForJobs);
-
-            var manifests = ReadManifestFiles(workingDirectory).ToList();
```

Comment:
All this code that gets resources from files has been moved to the `ResourceRetriever` class.